### PR TITLE
Metadata moved to separate jsons

### PR DESCRIPTION
**Pull request by skelly37**
# Description

*description of your pull request here*

# Contributor's checklist
Please mark as done everything appliable to the current state of your PR

## General
- [x] Newer functionality got documented (if needed, e.g. a new nupac enviroment variable)
- [x] Existing documentation is not outdated (e.g. changed inteface)
- [x] New code is tested
- [x] All files have up to data metadata
- [x] Pull request is not a draft (we have to know that you are ready for review)
## If submitting a module
- [ ] All exported functions have documentation added to them
- [ ] All exported functions have completion commands (unless non-feasible)
- [ ] All arguments to exported functions have documentation added to them
- [ ] All exported functions have examples added to them if usage is non obvious
- [ ] Helper functions and variables are not exported from the module
## If submitting a stand-alone script
- [ ] The script has a proper shebang at the top (`/usr/bin/env nu` should work in most circumstances)
- [ ] The script has proper file permissions (Run `chmod +x` on the file before push)

# Additional context

*do you have some questions, need some help or anything else?*

### CODE REVIEW

File: modules/nupac/nupac.nu [hunk 1]
```diff
@@ -210,18 +220,25 @@ def install-package [
     }
 
     print $"Installing ($package.name)"
+    mkdir (get-package-parent $package| into string)
     fetch ($package.raw-url | into string)
     |save (get-package-location $package | into string)
+    fetch ($package.raw-url | into string | str replace --string ".nu" ".json")
```
Comment (Yethal) [inline modules/nupac/nupac.nu:226]: wtf

File: installer.nu [hunk 2]
```diff
@@ -9,93 +9,96 @@ let nupac_repo_cache = $"https://raw.githubusercontent.com/skelly37/nupac/($defa
...
-    (scripts-path)
-    |path join "nupac"
+)
+let packages_path = (
+    $nupac_base_path
+    |path join 'packages'
 )
 
-# nupac index
+# # nupac index
```
Comment (skelly37) [inline installer.nu:26]:
```suggestion
# nupac index
```

File: installer.nu [hunk 3]
```diff
@@ -9,93 +9,96 @@ let nupac_repo_cache = $"https://raw.githubusercontent.com/skelly37/nupac/($defa
...
 )
 
 let nupac_json_path = (
-    (scripts-path)
-    |path join "nupac/nupac.json"
-    |into string
+    $nupac_module_path
+    |path join 'nupac.json'
 )
+$nu.scope.vars|select name value|sort-by value
```
Comment (skelly37) [inline installer.nu:49]: what's the point of it?

File: installer.nu [hunk 4]
```diff
@@ -9,93 +9,96 @@ let nupac_repo_cache = $"https://raw.githubusercontent.com/skelly37/nupac/($defa
...
+mkdir $nupac_base_path $packages_path $nupac_module_path
```
Comment (skelly37) [inline installer.nu:50]:
```suggestion
mkdir $nupac_module_path
```

No need to use mkdir thrice, as `$nupac_module_path` contains all of them

File: installer.nu [hunk 5]
```diff
@@ -9,93 +9,96 @@ let nupac_repo_cache = $"https://raw.githubusercontent.com/skelly37/nupac/($defa
...
+
+# #let actual_nupac_hash = (
+# #    open $nupac_path
+# #    |hash sha256
+# #)
+
+# #if ($expected_nupac_hash != $actual_nupac_hash) {
+# #    print $"Expected: ($expected_nupac_hash)"
+# #    print $"Actual:   ($actual_nupac_hash)"
+# #    rm -r (scripts_path)
```
Comment (skelly37) [inline installer.nu:92]: oh boy

File: modules/nupac/nupac.nu (hunk 1 above)
Comment (skelly37) [reply modules/nupac/nupac.nu:226]: since we store path to the .nu in the url, we can obtain it by a simple replacement

File: installer.nu (hunk 3 above)
Comment (Yethal) [reply installer.nu:49]: I used this for debugging, it's redundant now

File: installer.nu (hunk 4 above)
Comment (Yethal) [reply installer.nu:50]: I wanted to be verbose but good point

**Review by Yethal — APPROVED** [review]

**Pull request merged**